doc/developer/platform/formalism.md

```diff
@@ -114,6 +113,9 @@ after `F` are `{6, 7, 8, ...}`.
 
 The latest frontier is the empty set `{}`. No time is later than `{}`.
 
+A *stateful timeline* is a *timeline* and a *frontier* of the latest times of
```

Comment:
*stateful timeline*s are used earlier in the doc
> Since SQL does not
(generally) include an explicit model of time, the adapter picks timestamps for
SQL statements using a stateful *timeline* object.

However, it's never defined. I think this explanation provides a good abstraction into how we use timelines to pick timestamps.